### PR TITLE
My First PR

### PR DESCRIPTION
**Pull request by Fatema-Mohd**

### CODE REVIEW

File: Login UseCase/Login.java [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 // Data Layer
 // The User class represents a user with a username and password.
+
+
```
Comment (FatemaAlshaikh) [inline Login UseCase/Login.java:9]:
```suggestion
You Can Make Hash Passwords: Storing plain-text passwords is insecure. Hash passwords using a secure algorithm like BCrypt.
and you Make the username immutable if it's not expected to change.
```

File: Login UseCase/Login.java [hunk 2]
```diff
@@ -36,6 +34,8 @@ public void setPassword(String password) {
 
 // UserRepository manages user data in a simple in-memory map.
 // For storing and retrieving user information.
+
+
```
Comment (FatemaAlshaikh) [inline Login UseCase/Login.java:38]: You Can Check for duplicate users when adding.
You can Use a ConcurrentHashMap for better concurrency if accessed by multiple threads.

File: Login UseCase/Login.java [hunk 3]
```diff
@@ -80,6 +82,8 @@ public boolean authenticate(String username, String password) {
 // LoginController serves as the presentation layer for handling user interaction
 // for login operations. It interacts with the AuthenticationService to manage
 // user login attempts.
+
+
```
Comment (FatemaAlshaikh) [inline Login UseCase/Login.java:86]: You Can Return messages instead of printing directly for better UI integration.
You Can add Log authentication attempts.

File: Login UseCase/AuthenticationServiceTest.java [hunk 4]
```diff
@@ -1,10 +1,9 @@
 import org.junit.Before;
 import org.junit.Test;
-
 import static org.junit.Assert.*;
```
Comment (FatemaAlshaikh) [inline Login UseCase/AuthenticationServiceTest.java:4]: You Can Add tests for edge cases, such as duplicate usernames or invalid input formats.

File: Login UseCase/UserRepositoryTest.java [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import org.junit.Before;
 import org.junit.Test;
-
 import static org.junit.Assert.*;
```
Comment (FatemaAlshaikh) [inline Login UseCase/UserRepositoryTest.java:4]:
```suggestion
You Can Add a test for rejecting duplicate users.
```

File: Login UseCase/AuthenticationServiceTest.java (hunk 4 above)
Comment (FatemaAlshaikh) [inline Login UseCase/AuthenticationServiceTest.java:4]:
```suggestion
You Can Add tests for edge cases, such as duplicate usernames or invalid input formats
```

File: Login UseCase/Login.java (hunk 2 above)
Comment (FatemaAlshaikh) [inline Login UseCase/Login.java:54]:
```suggestion
You Can Check for duplicate users when adding.
You can Use a ConcurrentHashMap for better concurrency if accessed by multiple threads.
```

File: Login UseCase/Login.java (hunk 3 above)
Comment (FatemaAlshaikh) [inline Login UseCase/Login.java:106]:
```suggestion
You Can Return messages instead of printing directly for better UI integration.
You Can add Log authentication attempts.
```

File: Login UseCase/UserTest.java [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import org.junit.Test;
-
 import static org.junit.Assert.*;
 
+
```
Comment (FatemaAlshaikh) [inline Login UseCase/UserTest.java:4]:
```suggestion
You Can Include tests for hashing
```

File: Login UseCase/AuthenticationServiceTest.java [hunk 7]
```diff
@@ -1,10 +1,9 @@
...
 
-public class AuthenticationServiceTest {
```
Comment (FatemaAlshaikh) [inline Login UseCase/AuthenticationServiceTest.java:5]:
```suggestion
You Can Add tests for edge cases, such as duplicate usernames or invalid input formats
```

**Review by FatemaAlshaikh — CHANGES_REQUESTED** [review]
I hope you develop and improve the code .
Thank you .

**Review by FatemaAlshaikh — APPROVED** [review]
good work
